### DISCUSSION TITLE
Breaking Change ZWave JS UI (> 2024-11-29)

### DISCUSSION THREAD

**Discussion by MickLesk** [community-scripts/ProxmoxVE#635] [Guides]
On 2024-11-29 a breaking change was introduced in ZWave JS UI. (https://github.com/community-scripts/ProxmoxVE/pull/585)
ZWave JS UI is now operated via additional Envoirement values.

After these steps have been performed, it will be possible to simply execute the “update” statement again. 

Please note the following steps before updating:

1. Do an Backup of your installation:
```bash
cp -r /opt/zwave-js-ui /opt/zwave-js-ui_bak
```

2. Remove old folder & create new folder
```bash
rm -rf /opt/zwave-js-ui
mkdir -p /opt/zwave-js-ui
mkdir -p /opt/zwave_store
```

3. Move your old storage files to this path
```bash
cp -r /opt/zwave-js-ui_bak/store/*  /opt/zwave_store
```

4. Enter Main Folder:
```bash
cd /opt/zwave-js-ui
```

5. Update Package:
```bash
RELEASE=$(curl -s https://api.github.com/repos/zwave-js/zwave-js-ui/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
wget -q https://github.com/zwave-js/zwave-js-ui/releases/download/${RELEASE}/zwave-js-ui-${RELEASE}-linux.zip
unzip -q zwave-js-ui-${RELEASE}-linux.zip
```

6. Create .env File:
```bash
cat <<EOF >/opt/.env
ZWAVEJS_EXTERNAL_CONFIG=/opt/zwave_store/.config-db
STORE_DIR=/opt/zwave_store
EOF
```

7. Update Service
```bash
nano /etc/systemd/system/zwave-js-ui.service
``` 

Check [Service] and add or change missing:
```bash
User=root
WorkingDirectory=/opt/zwave-js-ui
ExecStart=/opt/zwave-js-ui/zwave-js-ui-linux
EnvironmentFile=/opt/.env
```

**Comment by xanderphillips** [comment]
Worked perfectly for me.  After second time of losing my store contents I searched and found this fix.  Thank you!  Would be awesome if the fix could be back-engineered into the updated version to prevent the loss of store folder for those who have not updated yet.

**Reply by GeorgeCastanza** [reply]
Thanks for this @MickLesk.

**Reply by tremor021** [reply]
Always backup your LXCs before issuing a update. You can never know if they changed something in the way app gets updated
